lib/sqlalchemy/dialects/oracle/types.py

```diff
@@ -214,6 +214,25 @@ def as_generic(self, allow_nulltype=False):
             day_precision=self.day_precision,
         )
 
+    def literal_processor(self, dialect):
+        def process(value):
+            if value is not None:
+                if value.days:
+                    value = (
+                        str(value).replace(" days,", "").replace(" day,", "")
+                    )
+                else:
+                    value = "0 " + str(value)
+                value = "INTERVAL '" + value + "' DAY"
+                if self.day_precision is not None:
+                    value += f"({self.day_precision})"
+                value += " TO SECOND"
+                if self.second_precision is not None:
+                    value += f"({self.second_precision})"
```

Comment:
Hi @CaselIT ,

I saw this method but wasn't sure how to specify precisions in this, but i have updated according to comment because i am realising that for literal query they might not need precision ig or if they require query could be something like `select round_method_for_respective_db_dialect(interval, precision_value) ...`.
have updated the query and tests accordingly

Reply:
If I'm not mistaken precision or length is not taken into consideration when using literal formatting.
For example literal formatter of `String(10)` will not do `value_str[:10]`, so it should be fine to not have precision taken into consideration here.

@zzzeek do you have a general rule here?

Reply:
Hi,

Found a corner case on this, if i insert a timedelta of 100 days. then it fails `leading_precision is the number of digits in the leading field. Accepted values are 0 to 9. The default is 2.` so if i have a timedelta of 100 days which has 3 digits it fails.

```py
sqlalchemy.exc.DatabaseError: (cx_Oracle.DatabaseError) ORA-01873: the leading precision of the interval is too small
[SQL: INSERT INTO t (x) VALUES (NUMTODSINTERVAL(8640000.0, 'SECOND'))]
```